x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -99,17 +102,63 @@ export type IndexPatternDimensionPanelProps = DatasourceDimensionPanelProps & {
   setState: (newState: IndexPatternPrivateState) => void;
 };
 
+function getOperationTypesForField({ type }: IndexPatternField): OperationType[] {
+  switch (type) {
+    case 'date':
+      return ['value', 'date_histogram'];
+    case 'number':
+      return ['value', 'sum', 'average'];
+    case 'string':
+      return ['value', 'terms'];
+  }
+  return [];
+}
+
+function getOperationResultType({ type }: IndexPatternField, op: OperationType): DataType {
+  switch (op) {
+    case 'value':
+      return type as DataType;
+    case 'average':
+    case 'count':
+    case 'sum':
+      return 'number';
+    case 'date_histogram':
+      return 'date';
+    case 'terms':
+      return 'string';
+  }
+}
+
 export function IndexPatternDimensionPanel(props: IndexPatternDimensionPanelProps) {
+  const [isOpen, setOpen] = useState(false);
+
   const fields = props.state.indexPatterns[props.state.currentIndexPatternId].fields;
-  const columns: IndexPatternColumn[] = fields.map((field, index) => ({
-    operationId: `${index}`,
-    label: `Value of ${field.name}`,
-    dataType: field.type as DataType,
+
+  const columns: IndexPatternColumn[] = fields
+    .map((field, index) => {
+      const validOperations = getOperationTypesForField(field);
+
+      return validOperations.map(op => ({
+        operationId: `${index}${op}`,
+        label: `${op} of ${field.name}`,
+        dataType: getOperationResultType(field, op),
+        isBucketed: op !== 'terms' && op !== 'date_histogram',
+
+        operationType: op,
+        sourceField: field.name,
+      }));
+    })
+    .reduce((prev, current) => prev.concat(current));
+
+  columns.push({
+    operationId: 'count',
```

Comment:
Shouldn't this id also be prefixed by the indexpattern id?

Reply:
The prefixes in the others are not indexpattern ID- but that should be part of this.